src/core/client/admin/components/UserStatus/BanModal.tsx

```diff
@@ -1,76 +1,115 @@
-              <Typography variant="header2" id="banModal-title">
-                Are you sure you want to ban{" "}
-                <strong>{username || <NotAvailable />}</strong>?
-              </Typography>
-            </Localized>
-            <Localized id="community-banModal-consequence">
-              <Typography>
-                Once banned, this user will no longer be able to comment, use
-                reactions, or report comments.
-              </Typography>
-            </Localized>
-          </HorizontalGutter>
-          <Flex justifyContent="flex-end" itemGutter="half">
-            <Localized id="community-banModal-cancel">
-              <Button variant="outlined" onClick={onClose}>
-                Cancel
-              </Button>
-            </Localized>
-            <Localized id="community-banModal-banUser">
-              <Button
-                variant="filled"
-                color="primary"
-                onClick={onConfirm}
-                ref={lastFocusableRef}
+  format,
+}) => {
+  function getMessage(): string {
+    return format("community-banModal-emailTemplate", {
+      username,
+    });
+  }
+  const [emailMessage, setEmailMessage] = useState(getMessage());
+  const [showMessage, setShowMessage] = useState(false);
+  return (
+    <Modal open={open} onClose={onClose} aria-labelledby="banModal-title">
+      {({ firstFocusableRef, lastFocusableRef }) => (
+        <Card className={styles.card}>
+          <CardCloseButton onClick={onClose} ref={firstFocusableRef} />
+          <HorizontalGutter size="double">
+            <HorizontalGutter>
+              <Localized
+                id="community-banModal-areYouSure"
+                strong={<strong />}
+                $username={username || <NotAvailable />}
               >
-                Ban User
-              </Button>
+                <Typography variant="header2" id="banModal-title">
+                  Are you sure you want to ban{" "}
+                  <strong>{username || <NotAvailable />}</strong>?
+                </Typography>
+              </Localized>
+              <Localized id="community-banModal-consequence">
+                <Typography>
+                  Once banned, this user will no longer be able to comment, use
+                  reactions, or report comments.
+                </Typography>
+              </Localized>
+            </HorizontalGutter>
+            <Localized id="community-banModal-customize">
+              <CheckBox
+                id="ban-edit-message"
+                name="ban-edit-message"
+                checked={showMessage}
+                onChange={e => {
+                  setShowMessage(e.target.checked);
+                }}
+              >
+                Customize ban email message
+              </CheckBox>
             </Localized>
-          </Flex>
-        </HorizontalGutter>
-      </Card>
-    )}
-  </Modal>
-);
 
-export default BanModal;
+            {showMessage && (
```

Comment:
Refer to https://github.com/coralproject/talk/blob/ce2c7c0d8f44f5eb35b6240497e080f42911727b/src/core/client/admin/routes/Configure/sections/Email/SMTP.tsx#L88 as an example of using `Field` with a subscription.